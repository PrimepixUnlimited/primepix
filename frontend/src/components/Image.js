import React from 'react';
import {Image} from 'react-native';

const ImageWrapper = ({image}) => <Image href={href} />;

export default ImageWrapper;
