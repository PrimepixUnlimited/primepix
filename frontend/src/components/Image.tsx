import React, { FC } from 'react'
import { Image, ImageStyle } from 'react-native'

interface Props {
  href?: string
  source?: { uri: string } | void
  style?: ImageStyle
}

const ImageWrapper: FC<Props> = ({ href, source, style }) => (
  <Image href={href} source={source} style={style} />
)

export default ImageWrapper
