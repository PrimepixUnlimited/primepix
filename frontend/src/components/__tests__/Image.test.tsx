// components/__tests__/Hello.tsx
import "react-native";
import * as React from "react";
import * as renderer from "react-test-renderer";

import Image from "../Image";

describe("Image snapshot", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(
        <Image source={{ uri: "../../assets/images/primepix-logo.png" }} />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
