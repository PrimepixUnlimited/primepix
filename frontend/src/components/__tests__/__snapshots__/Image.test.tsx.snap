// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image snapshot renders correctly 1`] = `
<Image
  source={
    Object {
      "uri": "../../assets/images/primepix-logo.png",
    }
  }
/>
`;
